ts/Stock/Indicators/DisparityIndex/DisparityIndexIndicator.ts

```diff
@@ -0,0 +1,282 @@
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/disparity-index
+     * @optionparent plotOptions.disparityindex
+     */
+    public static defaultOptions: DisparityIndexOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            /**
+             * The average used to calculate the Disparity Index indicator.
+             * By default it uses SMA. To use other averages, e.g. EMA,
+             * the stock/indicators/ema.js file needs to be loaded.
+             *
+             * If value is different than ema|dema|tema|wma, then sma is used.
+             */
+            average: 'sma',
+            index: 3,
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    } as DisparityIndexOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<DisparityIndexPoint> = void 0 as any;
+    public options: DisparityIndexOptions = void 0 as any;
+    public points: Array<DisparityIndexPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(this: DisparityIndexIndicator): void {
+        const args = arguments,
+            ctx = this, // Disparity Index indicator
+            params = args[1].params, // options.params
+            averageType = params && params.average ? params.average : void 0;
+
+        let averageIndicator;
+
+        switch (averageType) {
+        case 'ema':
+            averageIndicator = EMAIndicator;
+            break;
+        case 'dema':
+            averageIndicator = DEMAIndicator;
+            break;
+        case 'tema':
+            averageIndicator = TEMAIndicator;
+            break;
+        case 'wma':
+            averageIndicator = WMAIndicator;
+            break;
+        default: // use sma if any of the above strings do not match
+            averageIndicator = SMAIndicator;
+        }
+
+        ctx.averageIndicator = averageIndicator;
+
+        if (averageType !== 'sma') { // no need to check whether sma is loaded
+            // Check if the required average indicator modules is loaded
+            RequiredIndicatorMixin.isParentLoaded(
+                averageIndicator as any,
+                averageType,
+                ctx.type,
+                function (indicator: Highcharts.Indicator): undefined {
+                    indicator.prototype.init.apply(ctx, args);
+                    return;
+                }
+            );
+        }
+    }
+
+    public calculateDisparityIndex(
+        close: number,
```

Comment:
If this indicator allows changing the `index` of the price for a certain point in time, maybe use some more universal name like `currentPrice` for example?